trieste/acquisition/function.py

```diff
@@ -745,6 +745,121 @@ def ehvi_based_on_partitioned_cell(
+        """
+
+        tf.debugging.assert_positive(len(dataset), message="Dataset must be populated.")
+        mean, _ = model.predict(dataset.query_points)
+
+        _pf = Pareto(mean)
+        _reference_pt = get_reference_point(_pf.front)
+
+        sampler = BatchReparametrizationSampler(self._sample_size, model)
+
+        return batch_ehvi(sampler, self._jitter, _pf, _reference_pt)
+
+
+def batch_ehvi(
+    sampler, sampler_jitter: float, pareto: Pareto, reference_point: TensorType
+) -> AcquisitionFunction:
+
+    """
+    :param sampler: The posterior sampler, which given query points `at`, is able to sample
+        the possible observations at 'at'.
+    :param sampler_jitter: The size of the jitter to use in sampler when stabilising the Cholesky
+        decomposition of the covariance matrix.
+    :param pareto: a Pareto class instance containing the current obtained pareto points.
+    :param reference_point: The reference point for calculating hypervolume.
+    :return: The batch expected hypervolume improvement acquisition
+        function for objective minimisation.
+    """
+
+    def acquisition(at: TensorType) -> TensorType:
+        _batch_size = at.shape[-2]  # B
+
+        def gen_q_subset_indices(q: int) -> list:  # generate all subsets of [1, ..., q] as indices
+            indices = list(range(q))
+            return [tf.constant(list(combinations(indices, i))) for i in range(1, q + 1)]
+
+        samples = sampler.sample(at, jitter=sampler_jitter)  # [..., S, B, num_obj]
+
+        q_subset_indices = gen_q_subset_indices(_batch_size)
+
+        hv_contrib = 0.0
+        lb_points, ub_points = pareto.hypercell_bounds(
+            tf.constant([-inf] * samples.shape[-1], dtype=at.dtype), reference_point
+        )
+
+        def hv_contrib_on_samples(
+            obj_samples: TensorType,
+        ) -> TensorType:  # calculate samples overlapped area's hvi for obj_samples
+            # [..., S, Cq_j, j, num_obj] -> [..., S, Cq_j, num_obj]
+            overlap_vertices = tf.reduce_max(obj_samples, axis=-2)
+
+            overlap_vertices = tf.maximum(  # compare overlap vertices and lower bound of each cell:
+                tf.expand_dims(overlap_vertices, -3),  # expand a cell dimension
+                lb_points[tf.newaxis, tf.newaxis, :, tf.newaxis, :],
+            )  # [..., S, K, Cq_j, num_obj]
+
+            lengths_j = tf.maximum(  # get hvi length per obj within each cell
+                (ub_points[tf.newaxis, tf.newaxis, :, tf.newaxis, :] - overlap_vertices), 0.0
+            )  # [..., S, K, Cq_j, num_obj]
+
+            areas_j = tf.reduce_sum(  # sum over all subsets Cq_j -> [..., S, K]
+                tf.reduce_prod(lengths_j, axis=-1), axis=-1  # calc hvi within each K
+            )
+
+            return tf.reduce_sum(areas_j, axis=-1)  # sum over cells -> [..., S]
+
+        for j in range(1, _batch_size + 1):  # Inclusion-Exclusion loop
```

Comment:
Doesn't make a difference right now, but at some point we'll update the acquisition function so that it isn't retraced every time (I'm working on the infrastructure necessary for this at the moment).

```suggestion
        for j in tf.range(1, _batch_size + 1):  # Inclusion-Exclusion loop
```